lib/active_support/cache/database_store.rb

```diff
@@ -131,7 +138,7 @@ def write_multi_entries(hash, **_options)
       def entry_attributes(entry)
         expires_at = Time.zone.at(entry.expires_at) if entry.expires_at
 
-        { value: Marshal.dump(entry.value), version: entry.version.presence, expires_at: expires_at }
+        compression_attributes(entry.value).merge(version: entry.version.presence, expires_at: expires_at)
```

Comment:
pls use `update`, not `merge` here too. this will allocate a new object and there's no need